### DISCUSSION TITLE
Animations - Number "Flip"

### DISCUSSION THREAD

**Discussion by Fishbone69** [olikraus/u8g2#2109] [General]
Hi.  I am wanted to add a cool number flip animation to a clock like the one here: https://youtu.be/KrUJlZK1ct0

I was wondering how to do this in U8g2.  I could use a clip windows and scroll the numbers for the seconds and even the minutes since they are all lined up...but when you get to the hours, it wont work because the ":" symbol would flip.   I don't think multiple clip windows is supported.  Any other ideas?  (I love learning from all of the smart people on these discussions!)

Thanks!!

Fish

**Comment by olikraus** [comment]
You can change the clip window multiple times, like this: set clip window for hours, draw hours, set clip window for seconds and minutes, draw seconds and minutes.

**Reply by Fishbone69** [reply]
So you go back and forth scrolling a little each time?  (please confirm if that is the idea)  That will work if it is fast enough...maybe need an SPI rather than I2C.  By the way, I am new to this, but LOVE the work you did on the library...you should be proud!  

Fish

**Reply by olikraus** [reply]
U8g2 is double buffered. You will draw into uC memory, then later the bitmap is transferred to the display. All scrolling will be in sync and parallel.
